py/server/deephaven/experimental/iceberg.py

```diff
@@ -250,3 +251,80 @@ def adapter_aws_glue(
     except Exception as e:
         raise DHError(e, "Failed to build Iceberg Catalog Adapter") from e
 
+
+def adapter(
+        properties: Dict[str, str],
+        name: Optional[str] = None
+) -> IcebergCatalogAdapter:
+    """
+    Create an Iceberg catalog adapter for an Iceberg catalog created from configuration properties. These properties
+    map to the Iceberg catalog Java API properties and are used to create the catalog and file IO implementations.
+
+    The minimal set of properties required to create an Iceberg catalog are the following:
+    - `catalog-impl` or `type` - the Java catalog implementation to use. When providing `catalog-impl`, the
+            implementing Java class should be provided (e.g. `org.apache.iceberg.rest.RESTCatalog` or
+            `org.apache.iceberg.aws.glue.GlueCatalog`. Choices for `type` include `hive`, `hadoop`, `rest`, `glue`,
+            `nessie`, `jdbc`.
+    - `uri` - the URI of the catalog
+    - `io-impl` - the Java file IO implementation to use. Common choices are: `org.apache.iceberg.aws.s3.S3FileIO`
+            and `org.apache.iceberg.hadoop.HadoopFileIO`
+
+    Other common properties include:
+    - `warehouse` - the location of the data warehouse.
+    - `client.region` - the region of the AWS client.
+    - `s3.access-key-id` - the S3 access key for reading files.
+    - `s3.secret-access-key` - the S3 secret access key for reading files.
+    - `s3.endpoint` - the S3 endpoint to connect to.
+
+    Example usage #1 - REST catalog with an S3 backend (using MinIO):
+    ```
+    from deephaven.experimental import s3, iceberg
+
+    properties = {
+        "type" : "rest",
+        "uri" : "http://rest:8181",
+        "io-impl" : "org.apache.iceberg.aws.s3.S3FileIO",
+        "client.region" : "us-east-1",
+        "s3.access-key-id" : "admin",
+        "s3.secret-access-key" : "password",
+        "s3.endpoint" : "http://minio:9000"
+    }
+
+    adapter = iceberg.adapter_generic(name="generic-adapter", properties=properties);
+    ```
+
+    Example usage #2 - AWS Glue catalog:
+    ```
+    from deephaven.experimental import s3, iceberg
+
+    ## Note: region and credential information are loaded by the catalog from the environment
+    properties = {
+        "catalog-impl" : "org.apache.iceberg.aws.glue.GlueCatalog",
+        "uri" : "s3://lab-warehouse/sales",
+        "warehouse" : "s3://lab-warehouse/sales",
+        "io-impl" : "org.apache.iceberg.aws.s3.S3FileIO"
+    }
+
+    adapter = iceberg.adapter_generic(name="generic-adapter", properties=properties);
+    ```
+
+    Args:
+        properties (Dict[str, str]): the location of the warehouse.
```

Comment:
docstring does not seem very insightful given all of the details above.  Seems like way more than a "location".